broadinstitute/single_cell_portal_core
# Updating to scp-ingest-pipeline:1.23.2, adding cell_type__custom facet

#### BACKGROUND & CHANGES
This update adds support for using the `cell_type__custom` column in compliant metadata files while no longer needing a corresponding entry in `cell_type` or `cell_type__ontology_label`.  This will unblock users who cannot find corresponding analogs for special/custom cell types not listed in the [Cell](https://www.ebi.ac.uk/ols/ontologies/cl) ontology.

#### MANUAL TESTING
1. Download [valid_cell_type__custom_v2.2.1.txt](https://github.com/broadinstitute/scp-ingest-pipeline/blob/jlc_ctcustom_nodep/tests/data/annotation/metadata/convention/valid_cell_type__custom_v2.2.1.txt) locally
2. Pull this branch, run any pending migrations and start all services:
```
.bin/docker-compose-setup.sh
```
3. On the homepage, click the `cell type` facet and confirm that `cell type (custom)` also shows (you may not have data present there depending on the state of your database)
![Screen Shot 2022-11-29 at 4 20 05 PM](https://user-images.githubusercontent.com/729968/204650298-58bf3033-6ada-4703-b689-7d9f943c7cf2.png)

4. Create a new study, and upload valid_cell_type__custom_v2.2.1.txt as a metadata file and ensure that it passes ingest
